src/home/homework.c

```diff
@@ -1636,10 +1636,39 @@ static int home_unlock(UserRecord *h) {
         return 1;
 }
 
+static int save_original_stdout(FILE **ret) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        /* libcryptsetup in debug mode might end up logging to stdout. Which interferes with out own use of
```

Comment:
don't we plug it into our logging facility? via `cryptsetup_log_glue()`?

https://github.com/systemd/systemd/blob/main/src/shared/cryptsetup-util.c#L88

Reply:
We do. But we do this by setting the callbacks on crypt_device objects. But there are some calls where we invoke things with a NULL crypt_device object, and then things log bypass our callbacks, and end up on stdout in case of debugs log...

Reply:
Can't we set it up without a device too? There's static storage for the default https://gitlab.com/cryptsetup/cryptsetup/-/blob/master/lib/setup.c#L547